File: examples/python/ex_access.py
from wiredtiger import wiredtiger_open

# Connect to the database and open a session
conn = wiredtiger_open('WT_TEST', 'create')
session = conn.open_session()

# Create a simple table
session.create('table:T', 'key_format=S,value_format=S')

# Open a cursor and insert a record
cursor = session.open_cursor('table:T', None)

cursor.set_key('key1')
cursor.set_value('value1')
cursor.insert()

# Iterate through the records
cursor.reset()
for key, value in cursor:
    print('Got record: ' + key + ' : ' + value)

conn.close()

File: examples/java/com/wiredtiger/examples/ex_access.java
package com.wiredtiger.examples;
import com.wiredtiger.db.*;

public class ex_access {
    public static void main(String[] args) {
        Connection conn = wiredtiger.open("WT_HOME", "create");
        Session s = conn.open_session(null);
        s.create("table:t", "key_format=S,value_format=u");
        Cursor c = s.open_cursor("table:t", null, null);
        System.out.println("Key format: " + c.getKeyFormat());
        System.out.println("Value format: " + c.getValueFormat());
        try {
            c.addKeyFieldString("foo");
            c.addValueFieldByteArray("bar".getBytes());
            c.insert();
            c.reset();
            while (c.next() == 0)
                System.out.println("Got: " + c.getKeyFieldString());
        } catch (WiredTigerPackingException wtpe) {
        }
        conn.close(null);
    }
}
